fabric/colors.py

```diff
@@ -22,13 +22,26 @@
 If ``bold`` is set to ``True``, the ANSI flag for bolding will be flipped on
 for that particular invocation, which usually shows up as a bold or brighter
 version of the original color on most terminals.
+
+It is now possible to disable color printing completely by setting the environment
+variable FABRIC_DISABLE_COLORS.  The value doesn't matter as long as it exists.
 """
 
+import os
 
 def _wrap_with(code):
 
     def inner(text, bold=False):
         c = code
+
+        try:
+            os.environ.get("FABRIC_DISABLE_COLORS")
+        except NameError:
+            os.environ["FABRIC_DISABLE_COLORS"] = None
+
+        if os.environ.get("FABRIC_DISABLE_COLORS") is not None:
```

Comment:
Pro tip, you can actually condense all of this to simply:

``` python
if os.environ.get("FABRIC_DISABLE_COLORS", None):
    return text
```

This is because `os.environ` is a regular dict and `dict.get()` takes a 2nd param for a default value to apply if the key isn't in the dictionary. This is usually preferred over doing a try/except :)